### PR TITLE
Create Make.mach.nasa-aitken-rome

### PR DESCRIPTION
**Pull request by peeples**
Makefile for NASA's Aitken Rome nodes

### CODE REVIEW

**Comment by peeples** [comment]
Added my various commits to updating the regular Pleiades file while I was at it

**Review by jwise77 — APPROVED** [review]

File: src/enzo/Make.mach.nasa-aitken-rome [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#  respectively.  jobordner@ucsd.edu
+#
+# *** WARNING *** WARNING *** WARNING *** WARNING *** WARNING ***
+
+MACH_OPT_WARN        = 
+MACH_OPT_DEBUG       = -g -O0
+# Best for pleiades ivy bridge nodes.
+# See: goo.gl/A0IFMV
+MACH_OPT_HIGH        = -g -O2  -ip -ipo
+MACH_OPT_AGGRESSIVE  = -O3  -ip -ipo
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-aitken-rome:103]:
```suggestion
# Best for Aitken Rome nodes
# See: https://www.nas.nasa.gov/hecc/support/kb/preparing-to-run-on-aitken-rome-nodes_657.html
MACH_OPT_HIGH        = -g -O2  -ip -ipo -march=core-avx2
MACH_OPT_AGGRESSIVE  = -O3  -ip -ipo -march=core-avx2
```

File: src/enzo/Make.mach.nasa-aitken-rome [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+#-----------------------------------------------------------------------
+# Includes
+#-----------------------------------------------------------------------
+
+LOCAL_INCLUDES_MPI    = -I$(LOCAL_MPI_INSTALL)/include
+LOCAL_INCLUDES_HDF5   = -I$(LOCAL_HDF5_INSTALL)/include
+LOCAL_INCLUDES_PYTHON = -I$(LOCAL_PYTHON_INSTALL)/include/python2.7/ \
+                        -I$(LOCAL_PYTHON_INSTALL)/lib/python2.7/site-packages/numpy/core/include
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-aitken-rome:112]: Python 2 is now EOL and this variable not processed further so I suggest to remove it.
```suggestion
```

File: src/enzo/Make.mach.nasa-aitken-rome [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+# Libraries
+#-----------------------------------------------------------------------
+#
+
+LOCAL_LIBS_MPI    = -L$(LOCAL_MPI_INSTALL)/lib -lmpi -lmpi++
+LOCAL_LIBS_HDF5   = -L$(LOCAL_HDF5_INSTALL)/lib -lhdf5 -lz
+LOCAL_LIBS_LZ     = -L/u/mpeeples/miniconda3/lib -lz
+LOCAL_LIBS_GRACKLE = -L$(LOCAL_GRACKLE_INSTALL)/lib -lgrackle
+#LOCAL_LIBS_HDF4   = -L$(LOCAL_HDF4_INSTALL)/lib -lm -lmfhdf -lz -ldf -ljpeg     
+#LOCAL_LIBS_PYTHON  = $(LOCAL_PYTHON_INSTALL)/lib/python2.6/config/libpython2.6.a -lpthread -lutil
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-aitken-rome:129]: As above 
```suggestion
```

File: src/enzo/Make.mach.nasa-aitken-rome [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+LOCAL_LIBS_MACH = -lifcore -lifport
+
+MACH_LIBS         = $(LOCAL_LIBS_HDF5) $(LOCAL_LIBS_MACH) $(LOCAL_LIBS_GRACKLE) #$(LOCAL_LIBS_PYTHON)
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-aitken-rome:133]: As above
```suggestion
MACH_LIBS         = $(LOCAL_LIBS_HDF5) $(LOCAL_LIBS_MACH) $(LOCAL_LIBS_GRACKLE)
```

File: src/enzo/Make.mach.nasa-pleiades [hunk 5]
```diff
@@ -107,7 +110,8 @@ MACH_LDFLAGS  =
 
 MACH_OPT_WARN        = 
 MACH_OPT_DEBUG       = -g -O0
-# Best for pleiades ivy bridge nodes.
+# Best for pleiades ivy bridge, haswell, and broadwell nodes.
+# Does NOT work for the Aitken Rome nodes; use separate machine file for those
 # See: goo.gl/A0IFMV
 MACH_OPT_HIGH        = -g -O2 -xAVX -ip -ipo
 MACH_OPT_AGGRESSIVE  = -O3 -xAVX -ip -ipo
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-pleiades:117]:
```suggestion
MACH_OPT_HIGH        = -g -O2 -axCORE-AVX512,CORE-AVX2 -xAVX -ip -ipo
MACH_OPT_AGGRESSIVE  = -O3 -axCORE-AVX512,CORE-AVX2 -xAVX -ip -ipo
```

File: src/enzo/Make.mach.nasa-pleiades [hunk 6]
```diff
@@ -107,7 +110,8 @@ MACH_LDFLAGS  =
 
 MACH_OPT_WARN        = 
 MACH_OPT_DEBUG       = -g -O0
-# Best for pleiades ivy bridge nodes.
+# Best for pleiades ivy bridge, haswell, and broadwell nodes.
```
Comment (pgrete) [inline src/enzo/Make.mach.nasa-pleiades:113]:
```suggestion
# Best for pleiades ivy bridge, haswell, broadwell, skylake and cascade lake nodes.
```

**Review by pgrete — CHANGES_REQUESTED** [review]
I suggest to add architecture specific optimization flags and remove the explicit reference to the python libs.

**Comment by jwise77** [comment]
@peeples I fixed the testing in #190, and I merged in these changes to run the test suite on this PR. But I couldn't push the tags to your PR. Could you issue the command `git fetch https://github.com/jwise77/enzo-dev --tags` to bring in the new gold-standard-v15 tag for testing?

@pgrete I incorporated your recommended changes to the machine file. Could you approve?

Thanks!

**Comment by pgrete** [comment]
It looks like this PR is now mixed with changes from #190 so #190 should go in first to cleanup this PR.

**Review by pgrete — APPROVED** [review]
LGTM

**Pull request merged**